addons/website_slides/wizard/slide_channel_invite.py

```diff
@@ -81,26 +66,29 @@ def action_invite(self):
             email(s), rendering any template patterns on the fly if needed """
         self.ensure_one()
 
+        if not self.env.user.email:
+            raise UserError(_("Unable to post message, please configure the sender's email address."))
+
         mail_values = []
         for partner_id in self.partner_ids:
             slide_channel_partner = self.channel_id._action_add_members(partner_id)
             if slide_channel_partner:
-                mail_values.append(self._prepare_mail_values(slide_channel_partner))
+                mail_values.append(self._prepare_mail_values(slide_channel_partner, self.env.user.email_formatted, self.env.user.partner_id))
 
         # TODO awa: change me to create multi when mail.mail supports it
         for mail_value in mail_values:
             self.env['mail.mail'].sudo().create(mail_value)
 
         return {'type': 'ir.actions.act_window_close'}
 
-    def _prepare_mail_values(self, slide_channel_partner):
+    def _prepare_mail_values(self, slide_channel_partner, email_from, partner_id):
         """ Create mail specific for recipient """
         subject = self.env['mail.render.mixin']._render_template(self.subject, 'slide.channel.partner', slide_channel_partner.ids, post_process=True)[slide_channel_partner.id]
         body = self.env['mail.render.mixin']._render_template(self.body, 'slide.channel.partner', slide_channel_partner.ids, post_process=True)[slide_channel_partner.id]
         # post the message
         mail_values = {
-            'email_from': self.email_from,
-            'author_id': self.author_id.id,
+            'email_from': email_from,
```

Comment:
Now that I see it done, I think we don't even need to pass the values to the function.
```
'email_from': self.env.user.email_formatted,
'author_id': self.env.user.partner_id,
```
Since we only use contextual values, we can directly refer to them.
Sorry for the back and forth.